PHILOSOPHY.md

```diff
@@ -1,13 +1,15 @@
 # Philosophy
 
- * Simplicity of design.
+ * Simplicity of design - there should be one obvious way to do it.
  * The code is the spec.
```

Comment:
This is the only point I don't understand. If the code has to be `Evolvable`  as detailed in the last point, how can the code be the spec, if it is always evolving?

Specs, from my point of view, are the source of truth that makes it possible for two implementations to talk to each other. That cannot be the case if the canonical implementation is the spec, because it will change making it incompatible with new developments that will arise.

Reply:
I'm in agreement with you. If following the 'one obvious way' guideline causes confusion, I'm in favor of removing it.

Here's how I see it: The 'one obvious way' guideline works well alongside the idea of being adaptable. It doesn't mean the previous approach was wrong; it simply emphasizes our aim for clarity in the current context. When things change, having a flexible codebase lets us adjust without being stuck to an old set of rules. Our goal is to find a balance between staying compatible and being able to adapt quickly.

Reply:
@moul

I think there is some confusion here, @ajnavarro was referring to the point `The code is the spec` and how it relates to the last point that was introduced in this PR `Evolvable, debuggable, and future-proof codebase`

Reply:
Sorry Manfred for the confusion, I was talking about the following point:

> * The code is the spec.